src/main/java/com/cloudbees/jenkins/GitHubPushTrigger.java

```diff
@@ -235,6 +239,31 @@ public void writeLogTo(XMLOutput out) throws IOException {
         @Inject
         private transient GitHubHookRegisterProblemMonitor monitor;
 
+        @Inject
+        private transient SCMTrigger.DescriptorImpl scmTrigger;
+
+        private transient int maximumThreads = Integer.MIN_VALUE;
+
+        public DescriptorImpl() {
+            checkThreadPoolSize();
+        }
+
+        /**
+         * Update the {@link java.util.concurrent.ExecutorService} instance.
+         */
+        /*package*/
+        synchronized void checkThreadPoolSize() {
+            if (scmTrigger != null) {
+                int count = scmTrigger.getPollingThreadCount();
+                if (maximumThreads != count) {
+                    queue.setExecutors(
+                            (count == 0
+                                    ? Executors.newCachedThreadPool(threadFactory())
+                                    : Executors.newFixedThreadPool(maximumThreads, threadFactory())));
```

Comment:
When you call `queue.setExecutor(queue.getExecutor())` it will basically shutdown its own executor (which is why I stopped trying to steal SCMTrigger's queue because `setExecutor` shuts down the old one (without a no-op check beforehand... Nice eh))

So yes the admin will have 110 threads until the 100 previously executing tasks complete. No need to wait for GC